File: python/structures/test/search/graph/dfs_test.py
from . import graph_gen as generator

import structures.src.searches.graph.dfs as dfs


def test_traverse():
    root = generator.get_graph()

    expected = '21 39 27 30 33 36 16 11 84 18 63 14 51'

    result = dfs.traverse(root)

    assert result == expected

File: ruby/test/searches/graph/dfs_test.rb
require_relative '../../../src/searches/graph/dfs'
require_relative 'graph_gen'

require 'test_helper'

require 'minitest/autorun'
require 'minitest/spec'

describe BreadthFirst do
    describe 'when traversing a graph' do
        it 'should produce correct output' do
            root = GraphGeneration.generate()

            expected = '21 39 27 30 33 36 16 11 84 18 63 14 51'

            DepthFirst.traverse(root).must_equal expected
        end
    end
end
